// kaalin/dist/words/cyrillic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cyrillic = void 0;
class Cyrillic {
    constructor(text) {
        this.uppercases = [
            'А', 'Ә', 'Б', 'Д', 'Е', 'Ф', 'Г', 'Ғ', 'Ҳ', 'Х', 'Ы', 'И', 'Ж', 'К', 'Қ', 'Л', 'М', 'Н', 'Ң',
            'О', 'Ө', 'П', 'Р', 'С', 'Т', 'У', 'Ү', 'В', 'Ў', 'Й', 'З', 'Ш', 'Ц', 'Ч', ' '
        ];
        this.lowercases = [
            'а', 'ә', 'б', 'д', 'е', 'ф', 'г', 'ғ', 'ҳ', 'х', 'ы', 'и', 'ж', 'к', 'қ', 'л', 'м', 'н', 'ң',
            'о', 'ө', 'п', 'р', 'с', 'т', 'у', 'ү', 'в', 'ў', 'й', 'з', 'ш', 'ц', 'ч', ' '
        ];
        this.text = text;
    }
    getUppercases() {
        return this.uppercases;
    }
    getLowercases() {
        return this.lowercases;
    }
    isUpper() {
        return [...this.text].every(char => this.uppercases.includes(char));
    }
    isLower() {
        return [...this.text].every(char => this.lowercases.includes(char));
    }
    isDigit() {
        return /^\d+$/.test(this.text);
    }
    isAlpha() {
        return /^[\p{L}\d]+$/u.test(this.text);
    }
    swapCase() {
        return [...this.text]
            .map(char => this.uppercases.includes(char)
            ? char.toLowerCase()
            : this.lowercases.includes(char)
                ? char.toUpperCase()
                : char)
            .join('');
    }
    upper() {
        const upperMapping = new Map(this.lowercases.map((lc, i) => [lc, this.uppercases[i]]));
        return [...this.text].map(char => upperMapping.get(char) || char).join('');
    }
    lower() {
        const lowerMapping = new Map(this.uppercases.map((uc, i) => [uc, this.lowercases[i]]));
        return [...this.text].map(char => lowerMapping.get(char) || char).join('');
    }
}
exports.Cyrillic = Cyrillic;
