// kaalin/dist/words/latin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Latin = void 0;
class Latin {
    constructor(text) {
        this.text = text;
    }
    static getUppercases() {
        return this.uppercases;
    }
    static getLowercases() {
        return this.lowercases;
    }
    isUpper() {
        return [...this.text].every(char => Latin.uppercases.includes(char));
    }
    isLower() {
        return [...this.text].every(char => Latin.lowercases.includes(char));
    }
    isDigit() {
        return /^[0-9]+$/.test(this.text);
    }
    isAlpha() {
        return /^[a-zA-Z]+$/.test(this.text);
    }
    swapCase() {
        return [...this.text].map(char => char === char.toUpperCase() ? char.toLowerCase() : char.toUpperCase()).join('');
    }
    upper() {
        const upperMapping = Object.fromEntries(Latin.lowercases.map((lc, i) => [lc, Latin.uppercases[i]]));
        return [...this.text].map(char => upperMapping[char] || char).join('');
    }
    lower() {
        const lowerMapping = Object.fromEntries(Latin.uppercases.map((uc, i) => [uc, Latin.lowercases[i]]));
        return [...this.text].map(char => lowerMapping[char] || char).join('');
    }
}
exports.Latin = Latin;
Latin.uppercases = [
    "A", "Á", "B", "D", "E", "F", "G", "Д", "H", "X", "Í", "I", "J", "K", "Q", "L", "M", "N", "Ń",
    "O", "Ó", "P", "R", "S", "T", "U", "Ú", "V", "W", "Y", "Z", "Ш", "C", "Ch", " "
];
Latin.lowercases = [
    "a", "á", "b", "d", "e", "f", "g", "Ǵ", "h", "x", "ı", "i", "j", "k", "q", "l", "m", "n", "ń",
    "o", "ó", "p", "r", "s", "t", "u", "ú", "v", "w", "y", "z", "sh", "c", "ch", " "
];
